routes/remove_route.js

```diff
@@ -0,0 +1,16 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// remove a route from a list of routes.
+
+module.exports = function (routes, path) {
+  for (var i = 0; i < routes.length; ++i) {
+    var route = routes[i]
+    if (route.path === path) {
+      routes.splice(i, 1)
+      break
+    }
+  }
```

Comment:
IMO this should blow up at initialization time and prevent the server from starting, so it doesn't need to be one of the `errno`-bearing error classes.  We could e.g. add a `ConfigurationError` class for this purpose.